src/beagle_accessories.hpp

```diff
@@ -46,14 +46,14 @@ struct BeagleAccessories {
   // first derivative matrices
   std::vector<int> node_deriv_index_ = {0};
 
-  BeagleAccessories(int beagle_instance, bool rescaling, const Tree &tree)
+  BeagleAccessories(int beagle_instance, bool rescaling, const Node::NodePtr root)
```

Comment:
I like this type change. 

Elsewhere in the code (e.g. https://github.com/phylovi/libsbn/blob/master/src/libsbn.cpp#L127) I use "topology" as a variable name in a context like this and would prefer that here.